src/main/java/spoon/reflect/factory/CompilationUnitFactory.java

```diff
@@ -54,6 +60,89 @@ public CompilationUnit create() {
 		return cu;
 	}
 
+	private File getBaseDirectory() {
+		return this.factory.getEnvironment().getSourceOutputDirectory().getAbsoluteFile();
+	}
+
+	private File getModuleDirectory(CtModule module) {
+		return new File(this.getBaseDirectory(), module.getSimpleName() + File.separatorChar);
+	}
+
+	private File getPackageDirectory(CtPackage ctPackage) {
+		CtModule module = ctPackage.getParent(CtModule.class);
+		File baseDir;
+
+		if (module == null || module.isUnnamedModule() || factory.getEnvironment().getComplianceLevel() <= 8) {
+			baseDir = this.getBaseDirectory();
+		} else {
+			baseDir = this.getModuleDirectory(module);
+		}
+
+		return new File(baseDir, ctPackage.getQualifiedName().replace(CtPackage.PACKAGE_SEPARATOR_CHAR, File.separatorChar));
+	}
+
+	public CompilationUnit getOrCreate(CtPackage ctPackage) {
+		if (ctPackage.getPosition().getCompilationUnit() != null) {
```

Comment:
ctPackage.getPosition() may be null, we should check for null here too

Reply:
OK I'll change that.